skt/reporter.py

```diff
@@ -688,6 +688,10 @@ def __init__(self, cfg):
         self.headers = [headers.strip() for headers in
                         cfg['reporter']['mail_headers']]
         self.subject = cfg['reporter']['mail_subject']
+        self.smtp_url = 'localhost'
+        if cfg.get('smtp_url'):
+            self.smtp_url = cfg.get('smtp_url')
```

Comment:
Unless I'm missing overrides somewhere, `self.smtp_url = cfg.get('smtp_url', 'localhost')` can be used instead of those 3 lines.

Reply:
Yeah that didn't work.  I implemented the parser arg like you requested and it auto-sets it to None, which means the 'localhost' will never be used.  Setting it as a default in the parser blocks the config file from overriding it. :-/
Instead I am using: self.smtp_url = cfg.get('smtp_url') or 'localhost'
which maintains the spirit of what you wanted. :-)

Reply:
Right, those two ideas don't play well together. This was just a shorthand in case you didn't want to do the arg :) Ternary operators for assignments work just as fine. Will check the changes tomorrow!